src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,59 @@
+package duke.task;
+
+import java.time.LocalDate;
+
+/**
+ * Encapsulates the parent class for all types of tasks.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+    protected LocalDate date;
+
+    /**
+     * Creates a generic task with a description.
+     * @param description The task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    public String getStatusIcon() {
+        return isDone
+                ? "Y"
+                : "N";
```

Comment:
Maybe you can consider making "Y" and "N" as private final variables and then refer to them?